engine/modelc/src/modelimporter_jni.cpp

```diff
@@ -1013,6 +1006,20 @@ JNIEXPORT jobject JNICALL Java_ModelImporter_LoadFromBufferInternal(JNIEnv* env,
 
     dmModelImporter::DestroyScene(scene);
 
+    env->ReleaseByteArrayElements(array, file_data, JNI_ABORT);
+
+    return jscene;
+}
+
+JNIEXPORT jobject JNICALL Java_ModelImporter_LoadFromBufferInternal(JNIEnv* env, jclass cls, jstring _path, jbyteArray array, jobject data_resolver)
+{
+    dmLogDebug("Java_ModelImporter_LoadFromBufferInternal: env = %p\n", env);
+    dmJNI::SignalContextScope env_scope(env);
+
+    jobject jscene;
+    DM_JNI_GUARD_SCOPE_BEGIN();
+        jscene = LoadFromBufferInternal(env, cls, _path, array, data_resolver);
+    DM_JNI_GUARD_SCOPE_END(return 0;);
```

Comment:
Moved the code to an internal function, in order to make it easier  to wrap the code with guards (and also more readable).